### ISSUE TITLE
"Return to" function [Suggestion]

### ISSUE THREAD

**Issue by rwaltr** [simwir/Shuttle-Train#11]
Along with setting the Destination for a shuttle. set it as

   Train route:
   Playerrequest - wait a time
   set dest to "Shuttle Parking"

The idea being that if a shuttle is waiting too long at a player requested destination, the player has either:
1.Got distracted
2.Has been delivered to the destination.
3.Multiple shuttles are in the network, and someone accidentally sent two shuttles to two different locations. or the wrong location, while another shuttle reaches the correct location.

Therefore, the shuttle will return to "shuttle parking" where it is out of the way of the rail network and may also be refueled. Leaving all passenger or cargo stations open to other trains.

Preferably. This could be a check mark within the dest request menu. Incase someone did want the shuttle to stay there.

**Comment by stanfear** [comment]
seems easy enough, just set the schedule to have a second station set to be the shuttle parking
the difficulty will be to have the player select the name of the parking station (maybe using a table for the station selection UI with a hideable checkbox to set the parking ?)

**Comment by stanfear** [comment]
Also, if this is added, a shuttle should only be called if stoped at a parking station.

**Comment by stanfear** [comment]
@simwir, if you want, I can take a look at how to do that (I have an idea that could work)

**Comment by rwaltr** [comment]
I thought of some situations that might cause issues.
My lua skills are extremely rusty. But if you set the trains scheduled to
`Player Dest:wait 30
shuttlepark:wait max time`
This will set it as a loop and not as a one way dest.

this is more of an ease-of-life issue. but the amount of time you would want to wait for the player is different if the player is boarding the train or unborading the train. but here is some shitty pseudocode

```

if train unoccupied by player when call is made #set wait times
{wait time == 30;}
else{
wait time == 10;}
while waittime>deltatime
{
     if train is occupied{
     exit;} #stop the loop and program checking for destination 
}
set train dest to ==#shuttlepark and mark shuttle as free 
```

**Comment by simwir** [comment]
@stanfear You are more than welcome to take a look at it.
@Blackphidora Checking if the player is in the train would be pretty easy. And there is a event for a trains changing state, so you could check for that, and then see if it is at the parking station.

**Comment by ghost** [comment]
My request was similar so I thought to add it here. How about a "Home" button in the UI? This way there is a quick way to return after shuttling out to an outpost.  Happy to create it as a separate issue if that's better.

**Comment by stanfear** [comment]
@anup-ap I think a better option would be to bookmark a station to always have it first in the list (that way, you can bookmark any station you want, not just the "home" one

**Comment by ghost** [comment]
agreed, that works. anything that allows exemption from the filter would be an improvement to avoid extra clicks hopping back in the shuttle and heading home

**Comment by rwaltr** [comment]
@anup-ap @stanfear No. This will ruin the point of moving the train back to the home station. as currently trains are picked by which train is closest to the station, not the player. therefor it would be possible another train to be picked to head home instead of the tran that just dropped you off.

EDIT: oh. yall are mentioning a diffrent thing.

**Comment by stanfear** [comment]
ok, back to the original idea !
what do you think about rerouting trains to parking stations if there is one less than XX tiles away ?

**Comment by stanfear** [comment]
@anup-ap I've created another feature request, see #25

**Comment by rwaltr** [comment]
@stanfear I mean I guess. But it seems like it be a better idea as just a checkbox. let me elaborate.

Your main base may have enough sidings, wait ques. and rails to support an extra "useless"(no cargo) train. but what about outpost. especially is RSO. You have a base that's 3 mins away. only one station, single track, you have two situations.
One. you will be there shortly. and don't mind the delay of the tram while you are there.
   OR
You are using the outpost to quickly get to the edge of your territory. and you will be out for a longer time period (say. killing biters or building more) .
So now you at a delima. Do I force the player to make another train station? Do I just send the train off anyway, Or not move the train because a parking space is 10000 tiles away?

**Comment by phi1010** [comment]
One solution a player could use to prevent the shuttle train from leaving without creating and naming another station: The player could set a circuit-closed signal on the route.

**Comment by Kenz-420** [comment]
Can i suggest this be mentioned on the main forum and main github. i almost didn't download it because i was worried it would just clog up my tracks,

I'm not sure what the coding abilities allow, but maybe you can add an option that a circuit network effects the wait/return ratio... so different stops can have different wait times... like say whenever i travel to [Coal Station] that station has no circuit so the train leaves in [30] seconds.... but at [Base Station] the wait [60] seconds.... or you could have a power switch that turns the wait on/off.

Couldn't help with the station placement .. i enjoy the immersion of waiting on the train :D

**Comment by simwir** [comment]
@rainthegamer I no longer have time to maintain this mod. A new version is being maintained by folk:
 [shuttle train lite](https://mods.factorio.com/mods/folk/folk-shuttle).
